ModManager_Classes/Utils/FrameworkExtensions.cs

```diff
@@ -127,6 +127,47 @@ public static bool TryGetModOpNodes(this XmlDocument Document, String ModOpId, o
             return ModOps is not null && ModOps.Count > 0;
         }
     }
+
+    //https://stackoverflow.com/questions/4238345/asynchronously-wait-for-taskt-to-complete-with-timeout
+    public static class TaskExtensions
+    {
+        public static async Task<TResult> TimeoutAfter<TResult>(this Task<TResult> task, TimeSpan timeout)
+        {
+            using (var timeoutCancellationTokenSource = new CancellationTokenSource())
+            {
+
+                var completedTask = await Task.WhenAny(task, Task.Delay(timeout, timeoutCancellationTokenSource.Token));
+                if (completedTask == task)
+                {
+                    timeoutCancellationTokenSource.Cancel();
+                    return await task;  // Very important in order to propagate exceptions
+                }
+                else
+                {
+                    throw new TimeoutException("The operation has timed out.");
```

Comment:
I am curious, how would I do the onsuccess/onerror here?

Reply:
Otherwise, I am catching any exception over here https://github.com/anno-mods/iModYourAnno/blob/698f47136f3b224287afb111734c387c7fd584a5/ModManager_Classes/GithubIntegration/Download/GithubDownloader.cs#L56-L78

not just for the custom timeouts, but for HTTPClient Exceptions as well.